### PR TITLE
Break provider-aws up by service

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Relevant proposal: https://github.com/crossplane/crossplane/pull/3939
Depends on: https://github.com/upbound/upjet/pull/194

This PR proposes the following changes according to the proposal https://github.com/crossplane/crossplane/pull/3939:
- Subpackages belonging to each API group is produced. An example is: provider-aws-kms.
- ProviderConfig, ProviderConfigUsage and StoreConfig are part of a config package named provider-aws-config.
- The monolith package (containing all the CRDs and associated controllers) is still produced.
- Each produced package except for the monolith package has the `pkg.crossplane.io/provider-family` label in its package metadata.
- Each service package except for the config package declares a dependency to the config package.

It also introduces the `SUBPACKAGES` make variable so that it's possible to build the individual subpackages. An example for building is:
```shell
make -j 3 SUBPACKAGES="monolith config ec2" build.all
```
One can also use the same variable to push those (already built) packages:
```shell
make -j 3 SUBPACKAGES="monolith config ec2" BRANCH_NAME=main publish
```

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Successfully built and pushed the monolith, config and ec2 subpackages.

Also successfully provisioned and destroyed an `ec2.VPC` instance using the package `.../upbound/provider-aws-ec2:v0.34.0-rc.0.20.ga26202f1`.

### CODE REVIEW

File: package/crossplane-config.yaml
Comment (ulucinar) [inline package/crossplane-config.yaml:1]: Do we know of a lightweight templating tool that we can quickly integrate into our build pipelines?

Comment (turkenh) [reply package/crossplane-config.yaml:1]: We can use [gomplate](https://docs.gomplate.ca/). It is already available in the [build submodule](https://github.com/upbound/build/blob/292f958d2d97f26b450723998f82f7fc1767920c/makelib/local.mk#L82-L86) (we may want to move it out of local.mk though).

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/vpc.yaml"

File: package/crossplane-config.yaml
Comment (ulucinar) [reply package/crossplane-config.yaml:1]: Thanks @turkenh. Let's proceed as separate manifests for now and once we have the installation of `gomplate` refactored into somewhere like the `k8s_tools.mk`, then we can utilize the tool for templating the `crossplane.yaml`.

File: package/crossplane.yaml [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+      Available resources and their fields can be found in the [Upbound
+      Marketplace](https://marketplace.upbound.io/providers/upbound/provider-aws).
+      If you encounter an issue please reach out on support@upbound.io email
+      address. This is a subpackage for the {{ APIGROUP }} API group.
+    friendly-name.meta.crossplane.io: Provider AWS ({{ APIGROUP }})
     auth.upbound.io/group: aws.upbound.io
+spec:
+  dependsOn:
+    - provider: {{ XPKG_REG_ORGS }}/provider-aws-config
+      version: "*"
```
Comment (turkenh) [inline package/crossplane.yaml:25]: Wouldn't this bring and rc version as a dependency to a released version? Which is why I suggested `>=v0.0.0` in the first place.

**Review by turkenh — APPROVED** [review]

**Review by ezgidemirel — APPROVED** [review]
I have updated the platform-ref-aws configuration with the sub-packages and successfully installed it. There are 160 CRDs installed to the cluster and I could successfully create the cluster claim.

**Pull request merged**